src/digest/digest.rs

```diff
@@ -221,7 +221,7 @@ impl Clone for Context {
 /// # Examples:
 ///
 /// ```
-/// # #[cfg(any(feature = "use_heap", test))]
+/// # #[cfg(all(feature = "use_heap", test))]
```

Comment:
Good catch.

I believe this should just be `#[cfg(feature = "use_heap")]`, and same below. We don't need to condition this on `test` at all because that is handled by the test driver, right?